client/app/pages/query-snippets/edit.js

```diff
@@ -16,6 +24,24 @@ function SnippetCtrl($routeParams, $http, $location, toastr, currentUser, Events
       editor.$blockScrolling = Infinity;
       editor.getSession().setUseWrapMode(true);
       editor.setShowPrintMargin(false);
+
+      $scope.$watch(
```

Comment:
Instead of watching a variable, you can use `ng-change` in the template to explicitly trigger.
```
<input ... ng-change="changeVimMode($event)">
```

Reply:
If I do that, I can't access the editor...